ios-trivia-game/API/FirebaseClient.swift

```diff
@@ -265,6 +316,35 @@ class FirebaseClient {
         let path = "\(Constants.GAME_ROOM_TABLE_NAME)"
         return ref.child(path).childByAutoId()
     }
+    
+    // gets the ids of random questions
+    func getRandomQuestions(categoryId: Int, maxNumOfQuestions: Int, complete: @escaping ([Int]) -> (), onError: ((Error?) -> ())?) {
+        self.getQuestionsBy(categoryId: "\(categoryId)", complete: {(snapshot) in
+            var ids : [Int] = []
+            print ("val \(snapshot.value)")
+            
+            if let questions = snapshot.value as? NSDictionary {
+                let allQuestions = questions.allValues
+                for questionData in allQuestions {
+                    if let questionDict = questionData as? NSDictionary {
+                        let triviaQuestion = TriviaQuestion(dictionary: questionDict)
+                        ids.append(triviaQuestion.id!)
```

Comment:
Why store just the Ids?
Since you've already grabbed the questions already you might be able to save network calls back to Firebase by passing the question set around